pyproject.toml

```diff
@@ -55,15 +54,30 @@ docs = [
     "numpydoc",
     "tomli; python_version < '3.11'"
 ]
-
 # we can use self-references to simplify all
 all = [
     "nectarchain[test,docs,dev]",
 ]
 
+
+[tool.pytest.ini_options]
+addopts = "--ignore=src/nectarchain/user_scripts"
+
+
+
 [tool.setuptools.packages.find]
 where = ["src"]
 exclude = ["nectarchain._dev_version"]
 
 [tool.setuptools_scm]
 write_to = "src/nectarchain/_version.py"
+
+[tool.ruff.lint]
+select = ["E", "F", "W"]  # This ensures it mimics Flake8's error codes.
+exclude = ["build/", "dist/", "*.pyc"] 
+
+[tool.ruff]
+line-length = 88
+
+[tool.black]
+line-length = 88
```

Comment:
I don't get why an option is added here for `black`, could you please explain ? In CI, or locally on push, this is controlled through the pre-commit hooks.

Reply:
This is not used by the CI, this is for local use (with `black check` )